google/ggrc-core
# GGRC-5769 Improve the ACL propagation logic

## Perf notes:

- Adding a new custom role on ggrc_test_db on assessments (with 33k assessments) took around 4 minutes.

# TODO: 
 - [x] ticket id
 - [x] commit cleanup
 - [x] Migration for creating acl roles for existing objects
 - [x] Fix person unassignment.

# Dependencies

This PR is `on hold` until the dependencies are merged:
- [x] https://github.com/google/ggrc-core/pull/8523
- [x] ~~https://github.com/google/ggrc-core/pull/8429~~
- [x] ~~https://github.com/google/ggrc-core/pull/8440~~


# Issue description

Update ACL propagation logic according to the new spec.
The changes in this PR are just for partially normalizing the ACL table by moving people out into access_control_people table. This PR should handle all tickets under this epic, and their subtasks.

# Steps to test the changes

- Check the code for performance regressions (check for proper eager queries)
- Check basic rbac
- Check that all permission queries have been properly updated
- Sanity check over the entire app

# Solution description

- People were moved from ACL to access control people table.
- Acl table got base_id, that is always set, for easier permission queries.
- Permission queries and setters for ACL have been updated
- The setter for ACL was modified in such a way that the frontend API remains the same. This will have to be updated in the future but the current PR serves as the first step.


# Sanity checklist




# PR Review checklist
